src/LMS7002M_mcu_cal.c

```diff
@@ -1223,40 +1225,23 @@ LMS7002M_API int LMS7002M_mcu_calibration_dc_offset_iq_imbalance_rx(LMS7002M_t *
 
     LMS7002M_mcu_set_parameter(self, MCU_REF_CLK, clk);
     LMS7002M_mcu_set_parameter(self, MCU_BW, bw);
-    LMS7002M_mcu_run_procedure(self, MCU_FUNCTION_CALIBRATE_RX);
-    int ret = LMS7002M_mcu_wait(self, 10000);
-    if (ret != MCU_NO_ERROR) {
-        LMS7_logf(LMS7_ERROR, "MCU DC Rx calibration failed: %s", status_message(ret));
-        return -1;
-    }
-    return 0;
-}
-
-LMS7002M_API int LMS7002M_mcu_calibration_dc_offset_iq_imbalance_tx(LMS7002M_t *self, LMS7002M_chan_t channel, float clk, float bw)
-{
-    LMS7002M_set_mac_ch(self, channel);
-
-    if (LMS7002M_mcu_write_calibration_program(self)) {
-        LMS7_logf(LMS7_ERROR, "Failed to write MCU calibration program");
+    if (direction == LMS_RX && external_loopback == 0)
+        LMS7002M_mcu_run_procedure(self, MCU_FUNCTION_CALIBRATE_RX);
+    else if (direction == LMS_TX && external_loopback == 0)
+        LMS7002M_mcu_run_procedure(self, MCU_FUNCTION_CALIBRATE_TX);
+    else if (direction == LMS_RX && external_loopback != 0)
+        LMS7002M_mcu_run_procedure(self, MCU_FUNCTION_CALIBRATE_RX_EXTLOOPB);
+    else if (direction == LMS_TX && external_loopback != 0)
+        LMS7002M_mcu_run_procedure(self, MCU_FUNCTION_CALIBRATE_TX_EXTLOOPB);
+    else {
+        LMS7_logf(LMS7_ERROR, "Unsupported calibration direction and loopback");
         return -1;
     }
 
-    LMS7002M_mcu_set_parameter(self, MCU_REF_CLK, clk);
-    LMS7002M_mcu_set_parameter(self, MCU_BW, bw);
-    LMS7002M_mcu_run_procedure(self, MCU_FUNCTION_CALIBRATE_TX);
     int ret = LMS7002M_mcu_wait(self, 10000);
     if (ret != MCU_NO_ERROR) {
-        LMS7_logf(LMS7_ERROR, "MCU DC Tx calibration failed: %s", status_message(ret));
+        LMS7_logf(LMS7_ERROR, "MCU DC Rx calibration failed: %s", status_message(ret));
```

Comment:
This error message is a bit misleading. It is correct for `direction == LMS_RX`, but otherwise not. 
Maybe something like:
```c
dir_string = direction == LMS_RX ? "RX" : "TX";
LMS7_logf(LMS7_ERROR, "MCU %s calibration failed: %s", dir_string, status_message(ret));
```
I found the direction to be quite helpful.